core/src/nucleus/ribosome/lifecycle/mod.rs

```diff
@@ -0,0 +1,276 @@
+pub mod genesis;
+pub mod validate_commit;
+use action::ActionWrapper;
+use error::HolochainError;
+use hash_table::entry::Entry;
+use holochain_dna::zome::capabilities::ReservedCapabilityNames;
+use instance::Observer;
+use nucleus::{
+    call_zome_and_wait_for_result,
+    ribosome::{
+        lifecycle::{genesis::genesis, validate_commit::validate_commit},
+        Defn,
+    },
+    FunctionCall,
+};
+use num_traits::FromPrimitive;
+use std::{str::FromStr, sync::mpsc::Sender};
+
+// Lifecycle functions are zome logic called by HC actions
+// @TODO should each one be an action, e.g. Action::Genesis(Zome)?
+// @see https://github.com/holochain/holochain-rust/issues/200
+
+#[derive(FromPrimitive)]
+pub enum LifecycleFunction {
```

Comment:
Confusion between `lifecycle` and `callback`
Rename to `callback` or list only the functions that belongs the the lifecycle capability

Reply:
@ddd-mtl as i'm sure you picked up, i don't know how capabilities work, i cargo culted that in... still waiting on a pair session with @zippy to go over that but maybe you can run me through it too

we've been going back and forward on names like "hook", "zome function", etc.

callback doesn't seem right to me either because they aren't being passed around at runtime, let's have a quick straw poll in our next meeting to see if we can pick something

Reply:
I'm also not comfortable with `callback` and agree with what you said. I was just sticking with the names used in the Go version.